### ISSUE TITLE
Expose WebSocketExtension

### ISSUE THREAD

**Issue by btaens** [davidje13/websocket-express#5]
It's fairly common practice to extend WebSocket in TypeScript in the following manner:
```typescript
interface IdWebSocket extends WebSocket {
    id: string;
}
```
And then assign an id for them immediately at accept:
```typescript
const ws: IdWebSocket = await res.accept();
ws.id = uuidv4();
```
Or something akin to this, so it can later be tracked.
However, this plugin extends WebSocket with its own WebSocketExtension, and that interface is not exported, so anyone who tries to add their own properties to the WebSocket has to pretty much not use `nextMessage()`. If it were exposed it would be a simple matter:
```typescript
interface IdWebSocket extends WebSocket, WebSocketExtension {
    id: string;
}
```
Not to mention passing the WebSocket to other methods in other classes itself is an issue, due to not being able to type hint what it is we're passing.

**Comment by davidje13** [comment]
Sounds reasonable. I'll update it when I get some time - hopefully in the next few days.

In the meantime you can use a workaround like this:

```ts
interface IdWebSocket extends WebSocket {
    id: string;
    nextMessage(options?: { timeout?: number | undefined }): Promise<{ data: Buffer; isBinary: boolean }>;
}
```

---

As an aside, an alternative approach to your specific situation would be to use a `WeakMap`:

```ts
const myGlobalWeakMap = new WeakMap();

// ...later...

const ws = await res.accept();
myGlobalWeakMap.set(ws, uuidv4());

// ...later...

const currentId = myGlobalWeakMap.get(ws);
```

(which avoids having to add properties to an object which is "owned" by a library)

**Comment by davidje13** [comment]
Published 3.1.2 with this exported. You can now use the `ExtendedWebSocket` type:

```ts
interface IdWebSocket extends ExtendedWebSocket {
  id: string;
}
```
